src/IndentRainbow.Logic/Parser/ColorParser.cs

```diff
@@ -22,20 +22,42 @@ public static Brush[] ConvertStringToBrushArray(string colors, double opacityMul
             var colorCount = splitColors.Length;
             var brushes = new List<Brush>();
 
+            List<Color> c = new List<Color>();
+
             for (var i = 0; i < colorCount; i++)
             {
                 try
                 {
-                    var brush = new SolidColorBrush((Color)ColorConverter.ConvertFromString(splitColors[i]));
-                    double alphaOfBrush = (brush.Color.A);
-                    var color = brush.Color;
-                    color.A = (byte)Math.Floor(alphaOfBrush * opacityMultiplier);
-                    brush.Color = color;
-                    brushes.Add(brush);
+                    c.Add((Color)ColorConverter.ConvertFromString(splitColors[i]));
                 }
                 catch (FormatException) { }
             }
 
+            for (var i = 0; i < c.Count; i++)
+            {
+                Brush brush;
+                if (colorMode == 1)
+                {
+                    if (i + 1 < colorCount)
+                        brush = new LinearGradientBrush(c[i], c[i + 1], 0.0);
+                    else
+                    brush = new LinearGradientBrush(c[i], c[0], 0.0);
```

Comment:
I think you will need to add a reference to "WindowsBase", otherwise the project won't compile since it cannot find the type "Point" which LinearGradientBrush depends on.